File: apis/src/main/java/com/beat/apis/ticket/application/dto/TicketUpdateDetail.java
package com.beat.apis.ticket.application.dto;

import com.beat.apis.booking.application.dto.BookingStatusType;

import java.time.LocalDateTime;

public record TicketUpdateDetail(
	Long bookingId,
	String bookerName,
	String bookerPhoneNumber,
	Long scheduleId,
	int purchaseTicketCount,
	LocalDateTime createdAt,
	BookingStatusType bookingStatus,
	String scheduleNumber
) {
	public static TicketUpdateDetail of(
		Long bookingId,
		String bookerName,
		String bookerPhoneNumber,
		Long scheduleId,
		int purchaseTicketCount,
		LocalDateTime createdAt,
		BookingStatusType bookingStatus,
		String scheduleNumber) {
		return new TicketUpdateDetail(bookingId, bookerName, bookerPhoneNumber, scheduleId, purchaseTicketCount,
			createdAt, bookingStatus, scheduleNumber);
	}
}

File: apis/src/main/kotlin/com/beat/apis/ticket/api/request/TicketUpdateDetail.kt
package com.beat.apis.ticket.api.request

import com.beat.apis.booking.api.type.BookingStatusType
import java.time.LocalDateTime

data class TicketUpdateDetail(
    val bookingId: Long,
    val bookerName: String?,
    val bookerPhoneNumber: String?,
    val scheduleId: Long?,
    val purchaseTicketCount: Int?,
    val createdAt: LocalDateTime?,
    val bookingStatus: BookingStatusType,
    val scheduleNumber: String?,
)
